### PR TITLE
CA-29 put kingdom/building

### PR DESCRIPTION
**Pull request by mrkvcstms**
- changed TownhallLevelException, so it had to be included in method signatures
- changed the first update in upgradeBuilding to not save, so had to edit some tests
- plus the obvious ones 😄

### CODE REVIEW

File: src/main/java/com/greenfoxacademy/chamaaragonite/building/BuildingRepository.java [hunk 1]
```diff
@@ -5,5 +5,5 @@
 
 @Repository
 public interface BuildingRepository extends CrudRepository<Building, Long> {
-
+  Building findById(long id);
```
Comment (ZaitzeV16) [inline src/main/java/com/greenfoxacademy/chamaaragonite/building/BuildingRepository.java]: We don't need this explicitly. `findById` is a default in the `CrudRepository`

File: src/main/java/com/greenfoxacademy/chamaaragonite/building/BuildingServiceImpl.java [hunk 2]
```diff
@@ -74,5 +74,10 @@ public boolean isValidBuildingType(String buildingType) {
     return Arrays.stream(BuildingType.values())
         .anyMatch(building -> building.name().equals(buildingType.toUpperCase()));
   }
+
+  @Override
+  public Building findById(long id) {
+    return this.buildingRepository.findById(id);
```
Comment (ZaitzeV16) [inline src/main/java/com/greenfoxacademy/chamaaragonite/building/BuildingServiceImpl.java]: The default `findById` return an Optional<Building> so we will need to tell spring what to do when there is no data (e.g.: `.orElse(null)`

File: src/main/java/com/greenfoxacademy/chamaaragonite/exceptions/TownhallLevelException.java [hunk 3]
```diff
@@ -1,8 +1,9 @@
 package com.greenfoxacademy.chamaaragonite.exceptions;
 
-public class TownhallLevelException extends RuntimeException {
+public class TownhallLevelException extends Exception {
 
   public TownhallLevelException() {
+
```
Comment (ZaitzeV16) [inline src/main/java/com/greenfoxacademy/chamaaragonite/exceptions/TownhallLevelException.java]: unnecessary blank line

File: src/main/java/com/greenfoxacademy/chamaaragonite/dtos/BuildingResponseDTO.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+
+@Getter
+@Setter
+@NoArgsConstructor
+public class BuildingResponseDTO extends ResponseDTO {
+
```
Comment (ZaitzeV16) [inline src/main/java/com/greenfoxacademy/chamaaragonite/dtos/BuildingResponseDTO.java]: Missing regions

File: src/test/java/com/greenfoxacademy/chamaaragonite/kingdom/KingdomServiceImplTest.java [hunk 5]
```diff
@@ -174,10 +175,12 @@ public void createBuilding_ValidUsernameBuildingTypeInsufficientGold_Insufficien
 
     this.kingdomService.createBuilding("Mine", "Jozsika");
   }
+  // endregion CreateBuildingTests
```
Comment (Elaiden) [inline src/test/java/com/greenfoxacademy/chamaaragonite/kingdom/KingdomServiceImplTest.java]:
```suggestion


```

**Pull request merged**